purchase_work_acceptance/models/purchase.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    wa_count = fields.Integer(
+        compute='_compute_wa_ids',
+        string='WA count',
+        default=0,
+    )
+    wa_ids = fields.Many2many(
+        comodel_name='work.acceptance',
+        compute='_compute_wa_ids',
+        string='Work Acceptances',
+    )
+    wa_line_ids = fields.One2many(
+        comodel_name='work.acceptance.line',
+        inverse_name='purchase_line_id',
+        string="WA Lines",
+        readonly=True,
+    )
+
+    def _compute_wa_ids(self):
+        for order in self:
+            order.wa_ids = \
+                order.mapped('order_line').mapped('wa_line_ids').mapped('wa_id')
+            order.wa_count = len(order.wa_ids)
+
+    @api.multi
+    def action_view_wa(self):
+        self.ensure_one()
+        act = self.env.ref('purchase_work_acceptance.action_work_acceptance')
+        result = act.read()[0]
+        create_wa = self.env.context.get('create_wa', False)
+        result['context'] = {
+            'default_purchase_id': self.id,
+            'default_partner_id': self.partner_id.id,
+            'default_company_id': self.company_id.id,
+            'default_currency_id': self.currency_id.id,
+            'default_date_due': self.date_planned,
+            'default_wa_line_ids': [(0, 0, {
```

Comment:
Maybe you should only pass the lines with `qty != 0`

Reply:
Edited : https://github.com/OCA/purchase-workflow/pull/774/commits/74b8c5f27338d714143dfee5372105bf91e269cc